migration/migration.go

```diff
@@ -19,20 +21,38 @@ import (
 	"github.com/sourcegraph/conc/pool"
 )
 
+type schemaMetadata struct {
+	Version   uint64
+	Done      bool
+	LastState []byte
```

Comment:
Can we remove the `Done` field and rename `LastState` to `IntermediateState`? If `IntermediateState == nil`, then the db is consistent; otherwise, it's in the middle of a migration.